### ISSUE TITLE
[BUG] Backup gets stuck on Video

### ISSUE THREAD

**Issue by Klutronic** [alextran1502/immich#190]
**Describe the bug**
When backing up my pictures and videos, it almost finishes and gets stuck.  I have seen this on multiple videos.  Length doesn't seem to matter.  They are all from an iPhone 12 Pro Max, 4K, 3840x2160, Dolby Vision, 60 FPS.

**Task List**
[x] I have read thoroughly the README setup and installation instructions.
[x] If my setup is different, I have included my docker-compose file.
[x] I have included my redacted `.env` file.

###################################################################################
# Database
###################################################################################

DB_USERNAME=immich
DB_PASSWORD=<redacted>
DB_DATABASE_NAME=<redacted>

###################################################################################
# Upload File Config
###################################################################################

UPLOAD_LOCATION=/mnt/immich

###################################################################################
# JWT SECRET
###################################################################################

JWT_SECRET=<redacted>

###################################################################################
# MAPBOX
####################################################################################

# ENABLE_MAPBOX is either true of false -> if true, you have to provide MAPBOX_KEY
ENABLE_MAPBOX=false
MAPBOX_KEY=

###################################################################################
# WEB
###################################################################################

# This is the URL of your vm/server where you host Immich, so that the web frontend
# know where can it make the request to.
# For example: If your server IP address is 10.1.11.50, the environment variable will
# be VITE_SERVER_ENDPOINT=http://10.1.11.50:2283

VITE_SERVER_ENDPOINT=http://192.168.0.215:2283

[x] I have included information on my machine, and environment.

Docker version 20.10.12, build 20.10.12-0ubuntu2~20.04.1

4GB RAM
2 vCPUs

**To Reproduce**
1. Open Immich
2. Tap backup icon
3. Start Backup
4. Gets stuck on the first video it tries to upload

**Expected behavior**
All photos and videos get backed up

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.1
 - Docker: version 20.10.12, build 20.10.12-0ubuntu2~20.04.1

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro Max
 - OS: 15
 - Version 15.4.1

**Additional context**
I have removed the files that get stuck a few times to make sure it wasn't certain videos, but all of them end up stuck.

**Comment by alextran1502** [comment]
@Klutronic Can you help
Me take a screenshot of the video setting  **Setting > Camera > Video**? I realized that I've never recorded the video in 4K 60fps so there was no test on that. I will dig into this

**Comment by Klutronic** [comment]
Hey @alextran1502, sure thing:

[Screenshot](https://imgur.com/a/zOrOV8c)

**Comment by alextran1502** [comment]
@Klutronic I haven't been able to reproduce this issue on my end (iPhone 12 Pro). I used the same setting and record a 4K Dolby Vision 60fps footage and was able to upload the file. 

Is there any short footage from your phone that you can share so I can put it on my phone to try uploading?

**Comment by boffnoff** [comment]
@Klutronic if you checked the other open bugs you’d notice this is a duplicate of #54 

@alextran1502 I know you struggled to load my 7Gb video but I have the same iOS video settings applied

**Comment by alextran1502** [comment]
@boffnoff @Klutronic Do you guys use the Nginx setting in the repo or you have your own Nginx setting?

**Comment by Klutronic** [comment]
@alextran1502 I use whatever is in the repo.  If you tell me how to get the Xcode logs Ill grab whatever you need.

**Comment by Klutronic** [comment]
I have gotten the videos to upload successfully.  I wasn't connecting locally (I used Nginx Proxy Manager so I can could access/backup my immich instance from outside my network), but as soon as I used IP address of my server on my local network it uploaded just fine.  So I guess for now I can only backup on my local network.  Let me know if you want to test outside the local network at some point!

**Comment by alextran1502** [comment]
> I have gotten the videos to upload successfully. I wasn't connecting locally (I used Nginx Proxy Manager so I can could access/backup my immich instance from outside my network), but as soon as I used the local IP address of my server it uploaded just fine. So I guess for now I can only backup on my local network. Let me know if you want to test outside the local network at some point!

@Klutronic Thank you, this is an infrastructure issue, let's open a new one to focus on that. 

@boffnoff Did you also use the same setup as the comment above when backing up the video?

**Comment by boffnoff** [comment]
@alextran1502 I used the default Nginx included with your docker-compose. I dont have a mac so can't grab Xcode logs sorry😞

I've tried uploading the file I previously sent you whilst I'm on the local network and remotely (via portforwarding) and the app regardless

**Comment by alextran1502** [comment]
@boffnoff I will try to record a video with the same length and upload it to see if this is an issue with a large video file.

**Comment by Klutronic** [comment]
@boffnoff You can actually grab logs from your iPhone if you connect to a Windows machine through USB and use a took called iTools. [iTools](https://www.hexnode.com/mobile-device-management/help/obtain-ios-device-logs-using-mac-and-windows/).  The trial worked just fine for me to grab the logs.

**Comment by boffnoff** [comment]
Thanks @Klutronic for the headsup, the log file is pretty long and couldn't see a way to filter out all the background rubbish... @alextran1502 checkout the log here: [immich iTools log.txt](https://github.com/alextran1502/immich/files/8803222/immich.iTools.log.txt)

**Comment by Klutronic** [comment]
> @Klutronic Thank you, this is an infrastructure issue, let's open a new one to focus on that.

Are you ready to work on the "through a reverse proxy off the network" issue or is that something you want to wait on?  I can create the issue if you want me to.

**Comment by alextran1502** [comment]
Some update on this issue, this is probably a memory error when the app tries to read a large file into RAM. I will update once I've come up with a solution
